### PR TITLE
Feature/rebranding - Storage -> Data

### PR DESCRIPTION
**Pull request by yochaude**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Rebranding of the module from Storage to Data + normalization of some public methods/classes to align with the specs.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: apps/sasquatch/src/projectDependency/res/values/env.xml [hunk 1]
```diff
@@ -11,5 +11,5 @@
     <string name="api_url" tools:ignore="MissingTranslation">https://api-gateway-core-integration.dev.avalanch.es/v0.1</string>
     <string name="auth_config_url" tools:ignore="MissingTranslation">https://config-integration.dev.avalanch.es</string>
     <string name="rum_key" tools:ignore="MissingTranslation">0123456789abcdef0123456789abcdff</string>
-    <string name="storage_api_url" tools:ignore="MissingTranslation">https://token-exchange-mbaas-integration.dev.avalanch.es/v0.1</string>
+    <string name="data_api_url" tools:ignore="MissingTranslation">https://token-exchange-mbaas-integration.dev.avalanch.es/v0.1</string>
```
Comment (guperrot) [inline apps/sasquatch/src/projectDependency/res/values/env.xml]: this should be data_token_exchange_url with the new specs

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 2]
```diff
@@ -76,7 +76,7 @@
     /**
      * Current API base URL.
```
Comment (guperrot) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]:
```suggestion
      * Current token exchange base URL.
```

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/DocumentWrapper.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import com.microsoft.appcenter.data.Constants;
+import com.microsoft.appcenter.data.Utils;
+import com.microsoft.appcenter.data.exception.DataException;
+
+/**
+ * A document coming back from CosmosDB.
+ */
+public class DocumentWrapper<T> extends DocumentMetadata {
+
+
```
Comment (guperrot) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/DocumentWrapper.java]:
```suggestion
```

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/DefaultPartitions.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.microsoft.appcenter.data;
```
Comment (dhei) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/DefaultPartitions.java]:
```suggestion
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

package com.microsoft.appcenter.data;
```

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/TimeToLive.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.microsoft.appcenter.data;
```
Comment (dhei) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/TimeToLive.java]:
```suggestion
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */
package com.microsoft.appcenter.data;
```

**Pull request merged**